Google.Api.Gax.Grpc/Rest/RestChannel.cs

```diff
@@ -105,51 +117,97 @@ private async Task AddAuthHeadersAsync(HttpRequestMessage request, RestMethod me
             }
         }
 
-        private async Task<Metadata> ReadHeadersAsync(Task<HttpResponseMessage> httpResponseTask)
-        {
-            var httpResponse = await httpResponseTask.ConfigureAwait(false);
-            // TODO: This could be very wrong. I don't know what headers we should really return, and I don't know about semi-colon joining.
-            var metadata = new Metadata();
-            foreach (var header in httpResponse.Headers)
-            {
-                metadata.Add(header.Key, string.Join(";", header.Value));
-            }
-            return metadata;
-        }
+        private async Task<Metadata> ReadHeadersAsync(Task<ReadHttpResponseMessage> httpResponseTask) =>
```

Comment:
I haven't checked, but if this is only called in one place we should probably just inline it.